src/borg/testsuite/archiver/extract_cmd_test.py

```diff
@@ -579,20 +579,32 @@ def patched_setxattr_EACCES(*args, **kwargs):
 
 @pytest.mark.skipif(not is_darwin, reason="only for macOS")
 def test_extract_xattrs_resourcefork(archivers, request):
+
+    def darwin_get_birthtime_ns(path):
+        st = os.stat(input_path)
+        if hasattr(st, "st_birthtime_ns"):
+            return st.st_birthtime_ns
+        elif is_darwin_feature_64_bit_inode:
+            return get_birthtime_ns(path, follow_symlinks=False)
+        elif hasattr(st, "st_birthtime"):
+            return int(st.st_birthtime * 10**9)
+        else:
+            return None
```

Comment:
Instead of duplicating this, you could add it to `borg.helpers.fs` as `get_birthtime_ns` and call it from all places needing it.

The os.stat call needs to be removed for that and st, path, fd should be given as parameters.

Side note: we'll need another such helper in the future for windows as they are deprecating the usage of ctime for getting the file creation time on windows (in future one shall use birthtime there, too).